interfaces/builtin/auditd_support.go

```diff
@@ -0,0 +1,77 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const auditdSupportSummary = `allows control over the kernel audit system and access to files relevant to auditd`
+
+const auditdSupportBaseDeclarationPlugs = `
+  auditd-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const auditdSupportBaseDeclarationSlots = `
+  auditd-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const auditdSupportConnectedPlugSecComp = `
+# Description: Can use netlink to communicate with kernel audit system.
+bind
+socket AF_NETLINK - NETLINK_AUDIT
+`
+
+const auditdSupportConnectedPlugAppArmor = `
+# Description: Can use netlink to communicate with kernel audit system.
+network netlink,
+
+# CAP_AUDIT_CONTROL required to enable/disable kernel auditing, change auditing
+# filter rules, and retrieve auditing status and filtering rules.
+capability audit_control,
```

Comment:
The interface was previously the name for the interface, but Alfonso and I discussed after Alex's comment above, and agreed auditd-support was more fitting since things like `/run/auditd.{pid,state}` are very much auditd-specific, and are worth including in the interface instead of requiring system-files. Would you prefer `audit-control` or `auditd` instead of `auditd-support`?